### PR TITLE
fix 404 not found page with PKCE flow by closing previous netty channel

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
https://github.com/user-attachments/assets/58e7caf3-79e9-4d63-8f1c-a9beeae77101

https://github.com/user-attachments/assets/44444a58-f346-4139-a779-e54e0f7da785

https://github.com/user-attachments/assets/41320b0a-79aa-42f5-8d5e-fbdd9e62cc99

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt [hunk 1]
```diff
@@ -182,15 +182,8 @@ internal class ToolkitOAuthCallbackHandler : OAuthCallbackHandlerBase() {
...
-        // only handle if we're actively waiting on a redirect
-        if (!oauthService().hasPendingRequest()) {
-            return false
-        }
-
-        // only handle the /oauth/callback endpoint
-        return request.uri().trim('/').startsWith("oauth/callback")
-    }
+    // only handle the /oauth/callback endpoint
+    override fun isSupported(request: FullHttpRequest) = request.uri().trim('/').startsWith("oauth/callback")
```
Comment (rli) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt:186]: probably also want to restrict to HTTP GET

Comment (Will-ShaoHua) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt:186]: gotcha

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt [hunk 2]
```diff
@@ -183,10 +194,15 @@ internal class ToolkitOAuthCallbackHandler : OAuthCallbackHandlerBase() {
...
+         * following login attempts will keep seeing 404 not found page, which could be only resolved by (1) re-start the IDE (2) wait until the TCP connection
+         * between browsers and IDE is closed
+         */
         // only handle if we're actively waiting on a redirect
-        if (!oauthService().hasPendingRequest()) {
-            return false
-        }
+//        if (!oauthService().hasPendingRequest()) {
+//            return false
+//        }
```
Comment (Will-ShaoHua) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/sso/pkce/ToolkitOAuthService.kt:205]: confirmed that tcp connection will indeed be closed if cancel button is clicked

https://github.com/user-attachments/assets/3b86fec2-51b8-422c-8a02-1336b8cc1366

**Review by manodnyab — APPROVED** [review]

**Comment by rli** [comment]
Cleaner solution in https://github.com/aws/aws-toolkit-jetbrains/pull/4924

**Pull request closed without merging**